pkgs/applications/editors/micro/default.nix

```diff
@@ -1,40 +1,33 @@
-{ stdenv, buildGoPackage, fetchFromGitHub, installShellFiles }:
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "micro";
-  version = "2.0.7";
-
-  goPackagePath = "github.com/zyedidia/micro";
+  version = "2.0.8";
 
   src = fetchFromGitHub {
     owner = "zyedidia";
     repo = pname;
     rev = "v${version}";
-    sha256 = "07ck1a9arklic3p0z50wcg608cvpba1kljvlfb4fljr6jhv5cmkb";
-    fetchSubmodules = true;
+    sha256 = "1b51fvc9hrjfl8acr3yybp66xfll7d43412qwi76wxwarn06gkci";
   };
 
   nativeBuildInputs = [ installShellFiles ];
 
   subPackages = [ "cmd/micro" ];
 
-  buildFlagsArray = let t = "${goPackagePath}/internal/util"; in ''
-    -ldflags=
-      -X ${t}.Version=${version}
-      -X ${t}.CommitHash=${src.rev}
-  '';
+  vendorSha256 = "19iqvl63g9y6gkzfmv87rrgj4c4y6ngh467ss94rzrhaybj2b2d8";
 
-  goDeps = ./deps.nix;
+  buildFlagsArray = [ "-ldflags=-s -w -X github.com/zyedidia/micro/v2/internal/util.Version=${version} -X github.com/zyedidia/micro/v2/internal/util.CommitHash=${src.rev}" ];
```

Comment:
I am a bit curious. Are these flags something required in new versions of Go?

Reply:
Sorry for the late response. Golang uses `-ldflags` to [set variable values at compile time](https://blog.cloudflare.com/setting-go-variables-at-compile-time/), we use it to set the correct version string, and the [`-s -w` helps to reduce the binary size](https://blog.filippo.io/shrink-your-go-binaries-with-this-one-weird-trick/).
These flags are not required per-se, but if we don't include them, calling `micro --version` doesn't show the correct version.